packages/core/test/test_bed_spec.ts

```diff
@@ -248,6 +248,22 @@ describe('TestBed', () => {
     expect(hello.nativeElement).toHaveText('Hello World!');
   });
 
+  // https://github.com/angular/angular/issues/42734
+  it('should override a component which is declared in an NgModule which is imported as a `ModuleWithProviders`',
+     () => {
+       // This test verifies that an overridden component that is declared in an NgModule that has
+       // been imported as a `ModuleWithProviders` continues to have access to the declaration scope
+       // of the NgModule.
+       TestBed.resetTestingModule();
+       TestBed.configureTestingModule({imports: [{ngModule: HelloWorldModule}]});
```

Comment:
Could you call out that this is a "module with providers" import by putting it in a separate variable? Something like...

```suggestion
       const moduleWithProviders = {ngModule: HelloWorldModule}]};
       TestBed.configureTestingModule({imports: [moduleWithProviders]});
```